File: libs/document-processor/src/model/handlers/sequential-workflow-operation.ts
import { z } from 'zod';
import { blueNodeField } from '@blue-labs/language';
import { SequentialWorkflowOperationSchema as ConversationSequentialWorkflowOperationSchema } from '@blue-repository/types/packages/coordination/schemas/SequentialWorkflowOperation';

import { handlerContractBaseSchema } from '../shared/index.js';
import type {
  SequentialWorkflow,
  SequentialWorkflowStep,
} from './sequential-workflow.js';

const sequentialWorkflowOperationSchemaBase =
  ConversationSequentialWorkflowOperationSchema.merge(
    handlerContractBaseSchema,
  ).extend({
    steps: z.array(blueNodeField()).optional(),
  });

export type SequentialWorkflowOperation = SequentialWorkflow & {
  operation?: string;
  steps?: SequentialWorkflowStep[];
};

export const sequentialWorkflowOperationSchema =
  sequentialWorkflowOperationSchemaBase as z.ZodType<SequentialWorkflowOperation>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/SequentialWorkflowOperation.java
package blue.language.types.conversation;

import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

@TypeAlias("Conversation/Sequential Workflow Operation")
@TypeBlueId("CGdxkNjPcsdescqLPz6SNLsMyak6demQQr7RoKNHbCyv")
public class SequentialWorkflowOperation {
    public String operation;

    public SequentialWorkflowOperation operation(String operation) {
        this.operation = operation;
        return this;
    }
}
